src/Components/Components/src/ParameterView.cs

```diff
@@ -145,10 +145,13 @@ internal bool HasDirectParameter(string parameterName)
         return false;
     }
 
-    // It's internal because there isn't a known use case for user code comparing
-    // ParameterView instances, and even if there was, it's unlikely it should
-    // use these equality rules which are designed for their effect on rendering.
-    internal bool DefinitelyEquals(ParameterView oldParameters)
+    /// <summary>
+    /// Returns whether the parameters in this <see cref="ParameterView"/> are definitely equal to <paramref name="oldParameters"/>.
+    /// This is primarily used to detect whether a component's parameters have changed in a way that requires it to be re-rendered.
+    /// </summary>
+    /// <param name="oldParameters">The parameters to compare against.</param>
+    /// <returns><c>true</c> if the parameters are definitely equal, otherwise false.</returns>
+    public bool DefinitelyEquals(ParameterView oldParameters)
```

Comment:
We might be able to avoid adding a public API if we added a shared implementation of this method that relied on the `ParameterView` enumerator. However, we would also have to split up part of the implementation of the internal `ChangeDetection` class so that _some_ of it is shared (it references `IEventCallback`, which is internal). Since event callbacks can't be supplied to root components, it's fine if we skip that check for root components anyway. But it's a question of whether we want to jump through all those hoops to share as much of the implementation as possible while avoiding new public API, or if it's better to just make this API public.

Reply:
Don't make this public, make it shared code at this point.